### PR TITLE
Honor type style preferences on tuple literals

### PR DESCRIPTION
**Pull request by jcouv**
### Customer scenario
`UseImplicitType` on tuple literals should apply when the user preference applies recursively on all the tuple elements.
For example, on `(int a, string) s = (a: 1, "hello");` we'll offer `var` if we would have offered `var` on `int a = 1;` and `string b = "hello";`.

### Bugs this fixes
Fixes https://github.com/dotnet/roslyn/issues/11094

### Performance impact
Low. The change is limited to the `IsTypeApparentInAssignmentExpression` method, so that it handles tuples by checking the tuple elements.

### Is this a regression from a previous update?
No.

### How was the bug found?
Found during the development of the tuples language feature.

### CODE REVIEW

**Comment by sharwell** [comment]
> Low.

This needs to be confirmed via AnalyzerRunner before merging. Since the primary performance concern is the introduction of an allocation for the capturing lambda, the minimum test would be #23582.

File: src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs [hunk 1]
```diff
@@ -75,6 +75,14 @@ internal static class TypeStyleHelper
             CancellationToken cancellationToken,
             ITypeSymbol typeInDeclaration = null)
         {
+            // tuple literals
+            if (initializerExpression.IsKind(SyntaxKind.TupleExpression))
```
Comment (sharwell) [inline src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs]: :bulb:

```
initializerExpression.IsKind(SyntaxKind.TupleExpression, out TupleExpressionSyntax tupleExpression)
```

**Comment by jcouv** [comment]
@sharwell AnalyzerRunner doesn't seem to be in [master](https://github.com/dotnet/roslyn/tree/master/src/Tools). But the commits mentioned in the instructions went to master a month ago.
What branch do you recommend to use AnalyzerRunner?

**Comment by sharwell** [comment]
@jcouv As of this moment, the two pull requests indicated in the instructions are not yet merged. You'll need to manually merge them into your local master branch to have a starting point for running the tool.

**Comment by jcouv** [comment]
Then I'll wait

**Comment by jcouv** [comment]
From offline discussion with @sharwell, using for-loop instead of LINQ query would alleviate his concern. I'll update.

**Comment by jcouv** [comment]
Retargeted to dev15.6.x branch

**Comment by jcouv** [comment]
@sharwell for review. Thanks

**Comment by jcouv** [comment]
Thanks for review and offline clarification of concern, @sharwell. I hadn't understood the concern was allocation from captures. I need to read comments more carefully.

**Comment by jcouv** [comment]
@Pilchie  for approval for 15.6. 
Not sure if the ask-mode process changed last night. Please advise.

**Pull request merged**